docs/proposals-accepted/202107-protobuf-openapi-httpapi.md

```diff
@@ -0,0 +1,73 @@
+| type     | title                                                        | status   | menu               |
+| -------- | ------------------------------------------------------------ | -------- | ------------------ |
+| proposal | Expose Thanos APIs to OpenAPI/protobuf and expose on website and UIs | accepted | proposals-accepted |
+
+
+
+## **Expose Thanos APIs to OpenAPI/protobuf and expose on website and UIs**
+
+- Owners:
+  - [@Hangzhi](https://github.com/Hangzhi)
+- Related Tickets:
+  - Program Proposal: https://github.com/thanos-io/thanos/issues/4102
+  - Need for auto-documentation in Prometheus:
+    - https://github.com/prometheus/prometheus/issues/7192
+    - https://github.com/prometheus/prometheus/issues/5567
+- Other docs:
+  - A proposal of LFX mentorship: https://docs.google.com/document/d/1GeyFCk4QkVSrx4pQ4OSmOA1KE_FvmJRi_aq9hr2GoZQ/edit?usp=sharing
+
+This design doc is proposing a design for Thanos HTTP API defined in protobuf and OpenAPI.
+
+## **Motivations**
+
+In order to improve Thanos usage for users, we would like to define our HTTP APIs in protobuf/OpenAPI and expose those in the website. This would allow users to use tools for documentation, validation, type checking, and even code generation to use our APIs efficiently.

```

Comment:
```suggestion
In order to improve Thanos usage for users, we would like to define our HTTP APIs in protobuf/OpenAPI and expose those in our repository. This would allow users to use tools for documentation, validation, type checking, and even code generation to use our APIs efficiently.

Similarly we want reuse this work in Prometheus.

```

Reply:
For readability it would be good to expand on the following parts:
* `In order to improve Thanos usage for users` - what do we mean by improve here? What positive benefit will this bring to users?  
* `protobuf/OpenAPI` - as far as I am aware these are two different & non-interchangeable things?